extensions/azure/data-plane/data-factory/src/main/java/org/eclipse/dataspaceconnector/azure/dataplane/azuredatafactory/KeyVaultClient.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.azure.dataplane.azuredatafactory;
+
+import com.azure.security.keyvault.secrets.SecretClient;
+import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+
+/**
+ * Client for Azure Key Vault, wrapping the Azure SDK.
+ */
+class KeyVaultClient {
```

Comment:
nit: EDC does have an `AzureVault` extension , is it possible/feasible to reuse that instead of creating this client ?

Reply:
As discussed offline we will keep this one here as it seems more easier and simple.